### PR TITLE
chore(repository): BREAKING CHANGE remove support for HTTP-based repository API

### PR DESCRIPTION
**Pull request by jkowalski**
Now that we've released v0.16.0 this removes the repositrory API support over HTTP as described in #3716.

This removes flags:

* `--[no-]legacy-api` in `kopia server start` (defaulted to `false` in v0.16.0)
* `--[no-]grpc` in `kopia repository server connect` (defaulted to `true` in v0.16.0)

Note: This change is targeting v0.17.0 and is not meant to be merged merged immediately, in case we need to make hotfix patches on top of v0.16.0

### CODE REVIEW

File: tests/end_to_end_test/api_server_repository_test.go [hunk 1]
```diff
@@ -38,34 +38,24 @@ const (
...
-	testAPIServerRepository(t, []string{"--no-grpc", "--legacy-api"}, false, false)
+	testAPIServerRepository(t, true)
 }
 
 //nolint:thelper
-func testAPIServerRepository(t *testing.T, serverStartArgs []string, useGRPC, allowRepositoryUsers bool) {
+func testAPIServerRepository(t *testing.T, allowRepositoryUsers bool) {
 	ctx := testlogging.Context(t)
 
 	var connectArgs []string
```
Comment (redgoat650) [inline tests/end_to_end_test/api_server_repository_test.go:57]: I think this is now unused/empty. Worth removing?
```suggestion
```

**Review by redgoat650 — APPROVED** [review]
A few minor suggestions but looks good.

File: cli/command_server_start.go [hunk 2]
```diff
@@ -31,10 +31,9 @@ type commandServerStart struct {
 
 	serverStartHTMLPath string
 
-	serverStartUI                  bool
-	serverStartLegacyRepositoryAPI bool
-	serverStartGRPC                bool
-	serverStartControlAPI          bool
+	serverStartUI         bool
+	serverStartGRPC       bool
```
[the remark this answers is not in the document]
Comment (jkowalski) [reply cli/command_server_start.go:35]: this allows one to create UI-only server, without supporting repository API.

File: tests/end_to_end_test/api_server_repository_test.go (hunk 1 above)
Comment (jkowalski) [reply tests/end_to_end_test/api_server_repository_test.go]: done, also cleaned up below

**Review by julio-lopez — APPROVED** [review]

File: cli/command_server_start.go (hunk 2 above)
[the remark this answers is not in the document]
Comment (redgoat650) [reply cli/command_server_start.go:35]: Is it worth renaming the `--no-grpc` flag then to be more descriptive of the use case? `--ui-only` or `--[no-]repository-api` maybe?

**Pull request merged**